admin/class-boldgrid-backup-admin-auto-updates.php

```diff
@@ -0,0 +1,245 @@
+ * @author     BoldGrid <support@boldgrid.com>
+ */
+
+use Boldgrid\Library\Library\Plugin\Plugins;
+
+/**
+ * Class: Boldgrid_Backup_Admin_Auto_Updates.
+ *
+ * @since SINCEVERSION
+ */
+class Boldgrid_Backup_Admin_Auto_Updates {
+	/**
+	 * Settings.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	public $settings;
+
+	/**
+	 * Active Plugins.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	public $plugins = array();
+
+	/**
+	 * Themes.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	public $themes = array();
+
+	/**
+	 * Active Plugins.
+	 *
+	 * @since SINCEVERSION
+	 * @var Boldgrid_Backup_Admin_Core
+	 */
+	public $core;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function __construct() {
+
+		$this->set_settings();
+		$plugins       = new \Boldgrid\Library\Library\Plugin\Plugins();
+		$this->plugins = $plugins->getAllPlugins();
+		$this->themes  = new \Boldgrid\Library\Library\Theme\Themes();
+
+		add_filter( 'automatic_updater_disabled', '__return_false' );
+		add_filter( 'auto_update_plugin', array( $this, 'auto_update_plugins' ), 10, 2 );
+		add_filter( 'auto_update_themes', array( $this, 'auto_update_themes' ), 10, 2 );
+
+		add_action( 'admin_init', array( $this, 'auto_update_core' ) );
+	}
+
+	/**
+	 * Is Premium Done.
+	 *
+	 * @since SINCEVERSION
+	 * @return bool
+	 */
+	public function is_premium_done() {
+		$license        = new \Boldgrid\Library\Library\License();
+		$is_premium     = $license->isPremium( 'boldgrid-backup' );
+		$premium_plugin = 'boldgrid-backup-premium/boldgrid-backup-premium.php';
+		$active_plugins = (array) get_option( 'active_plugins', array() );
+		$premium_active = in_array( $premium_plugin, $active_plugins, true ) || is_plugin_active_for_network( $premium_plugin );
+		return ( $is_premium && $premium_active );
+	}
+
+	/**
+	 * Set Settings.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function set_settings() {
+		$boldgrid_backup_settings = get_site_option( 'boldgrid_backup_settings', array() );
+		if ( isset( $boldgrid_backup_settings['auto_update'] ) ) {
+			$this->settings = $boldgrid_backup_settings['auto_update'];
+		} else {
+			$this->settings = array(
+				'days' => 7,
+			);
+		}
+	}
+
+	/**
+	 * Maybe Update Plugin.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param string $slug Plugin Slug.
+	 * @return bool
+	 */
+	public function maybe_update_plugin( $slug ) {
+		$days_to_wait          = $this->settings['days'];
+		$plugin                = \Boldgrid\Library\Library\Plugin\Plugins::getActivePluginBySlug( $this->plugins, $slug );
+		$days_since_release    = $plugin->updateData->days; //phpcs:ignore WordPress.NamingConventions.ValidVariableName
+		$plugin_update_enabled = array_key_exists( $plugin->getFile(), $this->settings['plugins'] ) ? (bool) $this->settings['plugins'][ $plugin->getFile() ] : false;
+		// if premium, check the days since it was released, if not premium then this is true.
+		if ( $this->is_premium_done() ) {
```

Comment:
This is only checking to see if premium was enabled to know whether or not to compare the release date to the days setting. The premium functionality is the ability to set the days setting. However, with that in mind, i thought a little more into it, and realized that if they do not have the premium version, then the 'days' setting wouldn't be set at all. Therefore, I resolved this by removing the is_premium_done() check all together, and then setting the 'days' setting to a default of 0. 
If premium is enabled, then it will then be set to whatever they set it as. If premium is not enabled, the updates will run immediately since the days setting is 0.